schema2Markdown.js

```diff
@@ -0,0 +1,259 @@
+            // for each required property show the name
+            required.forEach((propertyName) => {
+                markdownContent += `* \`${propertyName}\`\n`;
+            });
+            markdownContent += '\n';
+        } 
+
+        if (ref) {
+            markdownContent += `ref: ${ref}\n\n`;
+        }
+
+        if (schema.examples) {
+            markdownContent += `## Examples\n\n`;
+            markdownContent += '```json\n';
+            markdownContent += JSON.stringify(schema.examples, null, 2);
+            markdownContent += '\n```';
+        }
+    }
+
+    const frontMatter = generateFrontMatter(objectName, schema.description);
+    console.log('frontMatter:', frontMatter);
+
+    const outputFileName = `./website/versioned_docs/version-${version}/${schemaFolderName}/schemas/${title.replace(/\s+/g, '')}.md`;
+
+    fs.outputFileSync(outputFileName, `---\n${yaml.dump(frontMatter)}\n---\n\n${markdownContent}`);
+
+    // objectName must not contain any spaces
+    if (objectName != null) {
+        return schemaFolderName + '/schemas/' + objectName.replace(/\s+/g, '');
+    }
+}
+
+function escape(text) {
+    let output = text;
+    output = output.replace(/>/g, '\\>');
+
+    return output;
+}
+
+function generateFrontMatter(title, description) {
+    return {
+        title: title + ' Schema',
+        description: description,
+        sidebar_label: title + ' Schema',
+    };
+}
+
+function processSchemaFile(schemaFile, schemaFolderName, version) {
+    const schemaData = fs.readJSONSync(schemaFile);
+
+    // if there is allOf, then it is an object
+    const allOfArray = schemaData.allOf;
+    let sidebarItems = [];
+    if (Array.isArray(allOfArray) && allOfArray.length > 0) {
+        sidebarItems.push(generateObjectMD(schemaData, null, schemaFolderName, version));
+    }
+    if (schemaData.definitions) {
+        for (const [objectName, objectDetails] of Object.entries(schemaData.definitions)) {
+            sidebarItems.push(generateObjectMD(objectDetails, objectName, schemaFolderName, version));
+        }
+    }
+
+    // return sidebarItems.flat().filter(item => !item.endsWith('undefined'));
+    return sidebarItems;
+}
+
+function parseSchemaFolder(schemaFolderName, version) {
+    // Read all files in the schema folder
+    const schemaFiles = fs.readdirSync("./schemas/"+schemaFolderName)
+    .filter(file => file.endsWith('.json'))
+    .map(file => path.join("./schemas/"+schemaFolderName, file));
+
+    // Process each schema file
+    let sidebarItems = [];
+    for (const schemaFile of schemaFiles) {
+        sidebarItems.push(processSchemaFile(schemaFile, schemaFolderName, version));
+    }
+
+    // filter out null values
+    return sidebarItems.flat().filter(item => item);
+}
+
+function main() {
+
+    const versions = ["1.0", "1.1", "1.2", "2.0", "2.1"];
+
+    versions.forEach(version => {
+
+        let sidebarObject = require(`./website/versioned_sidebars/version-${version}-sidebars.json`)
```

Comment:
Personally, I'd prefer not using `require`, just because it's caused me problems when I've managed system changes/upgrades in the past. I don't have anything specific to point t, though.

Instead, you could use fs-extra's `readJsonSync`, to match the later use of `outputJsonSync`